sqldelight-core/src/main/kotlin/com/squareup/sqldelight/core/compiler/model/NamedQuery.kt

```diff
@@ -59,14 +60,20 @@ data class NamedQuery(
       } else {
         compoundSelect = select.queryExposed().flatMap {
           val table = it.table
+          val leftJoin = it.joinOperator?.node?.findChildByType(SqliteTypes.LEFT) != null
           return@flatMap it.columns.map {
             var name = it.functionName()
             if (!namesUsed.add(name)) {
               if (table != null) name = "${table}_$name"
               while (!namesUsed.add(name)) name += "_"
             }
 
-            return@map it.type().copy(name = name)
+            return@map it.type().run {
+              copy(
+                  name = name,
```

Comment:
Now that this is inside `run`, isn't this a no-op?

Reply:
huh. yea. guess im not testing this well :D